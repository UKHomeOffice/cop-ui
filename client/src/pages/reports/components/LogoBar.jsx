import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-navi';
import { useTranslation } from 'react-i18next';
import './LogoBar.scss';

const LogoBar = ({ setFullscreen }) => {
  const { t } = useTranslation();

  return (
    <div className="logo-bar">
      <p>
        {t('pages.reports.logo-bar.intro')}{' '}
        <Link href="/reports/">{t('pages.reports.logo-bar.link')}</Link>{' '}
        {t('pages.reports.logo-bar.outro')}
      </p>
      <button type="button" onClick={setFullscreen}>
        {t('pages.reports.logo-bar.fullscreen')}
      </button>
    </div>
  );
};

LogoBar.propTypes = {
  setFullscreen: PropTypes.func.isRequired,
};

export default LogoBar;
