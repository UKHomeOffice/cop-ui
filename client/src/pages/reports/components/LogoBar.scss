@import '~govuk-frontend/govuk/settings/all';

.logo-bar {
  align-items: center;
  bottom: -35px;
  display: flex;
  font-family: $govuk-font-family-nta;
  font-size: 12px;
  height: 35px;
  position: absolute;
  right: 0;

  button {
    border: none;
    background-color: transparent;
    color: govuk-colour('dark-grey');
    cursor: pointer;
    font-size: smaller;
    font-weight: $govuk-font-weight-bold;
    padding: 6px;
    text-transform: uppercase;
    &:focus {
      outline: none;
    }
  }

  p {
    color: govuk-colour('dark-grey');
    margin: 0;

    @media only screen and (max-width: 960px) {
      display: none;
    }
  }

  button,
  p {
    &:not(:last-child):after {
      content: '|';
      margin-left: 6px;
    }
  }
}
