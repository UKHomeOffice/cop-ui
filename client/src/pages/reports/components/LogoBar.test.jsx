import React from 'react';
import { shallow } from 'enzyme';
import LogoBar from './LogoBar';

describe('LogoBar', () => {
  const props = {
    setFullscreen: jest.fn(),
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('fires setFullscreen on button click', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    wrapper.find('button').simulate('click');
    expect(props.setFullscreen).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
