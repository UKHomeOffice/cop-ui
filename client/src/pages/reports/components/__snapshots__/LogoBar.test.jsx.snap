// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoBar matches snapshot 1`] = `
<div
  className="logo-bar"
>
  <p>
    pages.reports.logo-bar.intro
     
    <Link
      href="/reports/"
    >
      pages.reports.logo-bar.link
    </Link>
     
    pages.reports.logo-bar.outro
  </p>
  <button
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="button"
  >
    pages.reports.logo-bar.fullscreen
  </button>
</div>
`;
